sdk/python/kfp/_client.py

```diff
@@ -707,7 +707,7 @@ def __init__(self, spec, resource_references):
       pipeline_json_string = json.dumps(pipeline_obj)
     api_params = [kfp_server_api.ApiParameter(
         name=sanitize_k8s_name(name=k, allow_capital_underscore=True),
-        value=str(v)) for k,v in params.items()]
+        value=str(v) if type(v) not in (list, dict) else json.dumps(v)) for k,v in params.items()]
```

Comment:
What if we always do `json.dumps` regardless of the type?

I can think of the difference for boolean values:
`json.dumps(True) == 'true'`, while `str(True) == 'True'`. 

But I assume our api server should be able to handle both value correctly. Can you help verify that? Thanks!

Reply:
Okay, I also recall that `Enum` is by default not serializable. So while `json.dumps()` won't work on enum values but `str()` will.

Reply:
Also, if a list/dict has already been serialized by the user into a str, we would not want to re-serialize that value. That caused some issues for me while testing. This way we keep some backwards capability.